main/autogen/subclasses.cc

```diff
@@ -52,18 +65,8 @@ optional<Subclasses::Map> Subclasses::listAllSubclasses(core::Context ctx, Parse
             continue;
         }
 
-        // Get fully-qualified parent name as string
-        string parentName =
-            fmt::format("{}", fmt::map_join(ref.resolved.nameParts,
-                                            "::", [&ctx](const core::NameRef &nm) -> string { return nm.show(ctx); }));
-
-        // Add child class to the set identified by its parent
-        string childName =
-            fmt::format("{}", fmt::map_join(pf.showFullName(ctx, defn),
-                                            "::", [&ctx](const core::NameRef &nm) -> string { return nm.show(ctx); }));
-
-        auto &mapEntry = out[parentName];
-        mapEntry.entries.insert(make_pair(childName, defn.data(pf).type));
+        auto &mapEntry = out[ref.sym];
+        mapEntry.entries.insert({defn.data(pf).sym, defn.data(pf).type});
```

Comment:
Sorry for the back-and-forth, but we can do even better. `type` is stored on `sym`, because you can check `isClassOrModule()` on it.

Reply:
So SymbolRefs have
```
    enum class Kind : uint8_t {
        ClassOrModule = 0,
        Method = 1,
        FieldOrStaticField = 2,
        TypeArgument = 3,
        TypeMember = 4,
    };
 ```
 but it does not distinguish between classes or modules, that is inferred here https://github.com/sorbet/sorbet/blob/3a7d0cf064c03921774c672461b6b9593451fc19/main/autogen/autogen.cc#L75

Reply:
Oh nm, we need to disambiguate further here. Sorry. My bad.

Reply:
@youssefsoli I think there's `isClass` / `isModule` methods on `sym.data(ctx)` which will do this.